boefjes/boefjes/config.py

```diff
@@ -1,44 +1,94 @@
-from enum import Enum
+import logging
+import os
 from pathlib import Path
-from typing import Optional
+from typing import Any, Dict, Optional, Tuple
 
-from pydantic import AnyHttpUrl, BaseSettings, PostgresDsn
+from pydantic import AmqpDsn, AnyHttpUrl, BaseSettings, Field, IPvAnyAddress, PostgresDsn
+from pydantic.env_settings import SettingsSourceCallable
 
+BASE_DIR: Path = Path(__file__).parent.resolve()
 
-class RuntimeConfiguration(Enum):
-    LOCAL = "local"
+# Set base dir to something generic when compiling environment docs
+if os.getenv("DOCS"):
+    BASE_DIR = Path("../")
 
 
-class Settings(BaseSettings):
-    base_dir: Path = Path(__file__).parent.resolve()
-    log_cfg: Path = Path(__file__).parent / "logging.json"
-
-    # Worker configuration
-    pool_size: int = 2
-    poll_interval: float = 10.0
-    worker_heartbeat: float = 1.0
-
-    # Queue configuration
-    queue_uri: str = "amqp://"
+class BackwardsCompatibleEnvSettings:
+    backwards_compatibility_mapping = {
+        "LOG_CFG": "BOEFJES_LOG_CFG",
+    }
 
-    # Runtime configuration
-    runtime: RuntimeConfiguration = RuntimeConfiguration.LOCAL
+    def __call__(self, settings: BaseSettings) -> Dict[str, Any]:
+        d: Dict[str, Any] = {}
+        env_vars = {k.lower(): v for k, v in os.environ.items()}
+        env_prefix = settings.__config__.env_prefix.lower()
 
-    katalogus_db_uri: PostgresDsn = "postgresql://xx:xx@host:5432/katalogus"
+        for old_name, new_name in self.backwards_compatibility_mapping.items():
+            old_name, new_name = old_name.lower(), new_name.lower()
 
-    scheduler_api: AnyHttpUrl = "http://localhost:8004"
+            # New variable not explicitly set through env,
+            # ...but old variable has been explicitly set through env
+            if new_name not in env_vars and old_name in env_vars:
+                logging.warning("Deprecation: %s is not valid, use %s instead", old_name.upper(), new_name.upper())
```

Comment:
Small nitpick, that isn't a blocker for merging: I would say "is deprecrated" instead of "is not valid", because it will still be valid as long was we keep the backwards compatibility.